### ISSUE TITLE
Insert guard pages below per-CPU stacks

### ISSUE THREAD

**Issue by abrestic-rivos** [rivosinc/salus#240]
@rajnesh-kanwal encountered a panic in #239 that turned out to be due to a stack overflow on one CPU scribbling over the `PerCpu` struct of another CPU. We should really be inserting guard pages between the CPU stacks so that we can get an easy to diagnose page-fault.

**Comment by glg-rv** [comment]
[Following discussion with @abrestic-rivos]

In general, this should be done in two planes. Inserting stack pages and possibly using something as probestack: https://docs.rs/compiler_builtins/latest/compiler_builtins/probestack/index.html which is unfortunately mostly handwritten x86(_64) assembly depending on a LLVM functionality architecture dependent and only available ATM for x86(_64).

The underlying problem is that a guard page is not enough, because especially in Rust code, with all the aggressive inlining and the stack-prone nature of the language, the prologue of a function very often is increased by more than two pages.

So, while technically inserting a few guard pages is enough, (U-mode is a bit more extreme on this: https://github.com/rivosinc/salus/blob/860457dd8d563228b59b98ec072194fe36cd9a2a/lds/umode.lds#L36), and it should be considered acceptable in the short term, long term solution will require something that should guarantee us that the stack has a limit.

**Comment by abrestic-rivos** [comment]
Also noting here that we probably should do something like Linux's `VMAP_STACK` where the stack is remapped to a different VA rather than using the identity map to access the stack. That way we can keep the stack page allocations physically contiguous while still being able to insert as many guard pages as we want.

**Comment by glg-rv** [comment]
Yes. As we now have per-CPU page tables, a CPU stack should also be mapped only in the CPU of interest. Having a fixed VA will help make things cleaner.
